analyses/cms-open-data-ttbar/utils/__init__.py

```diff
@@ -48,7 +52,20 @@ def set_style():
     plt.rcParams['text.color'] = "222222"
 
 
-def construct_fileset(n_files_max_per_sample, use_xcache=False, af_name=""):
+# If local_data_cache is a writable path, this function will download any missing file into it and
+# then return file paths corresponding to these local copies.
+def construct_fileset(n_files_max_per_sample, use_xcache=False, af_name="", local_data_cache=None):
+    if af_name == "ssl-dev":
+        if use_xcache:
+            raise RuntimeError("`use_xcache` and `af_name='ssl-dev'` are incompatible. Please only use one of them.")
+        if local_data_cache is not None:
+            raise RuntimeError("`af_name='ssl-dev'` and `local_data_cache` are incompatible. Please only use one of them.")
```

Comment:
Perhaps could degrade this to a logged warning?

Reply:
I was just cutting out edge cases: if we allow using ssl-dev together with local_data_cache, then we have to deal with the case in which they are both on explicitly: I have to replace `/data/alheld/` with the local path instead of replacing `https://xrootd-local.unl.edu:...`.

I can do that if you want of course, but I thought it would be simpler to just allow either one or the other.